pkg/gadgets/helpers.go

```diff
@@ -100,3 +109,88 @@ func IPStringFromBytes(ipBytes [16]byte, ipType int) string {
 		return ""
 	}
 }
+
+var timeDiff time.Duration
+
+func init() {
+	var t unix.Timespec
+	err := unix.ClockGettime(unix.CLOCK_BOOTTIME, &t)
+	if err != nil {
+		panic(err)
+	}
+	timeDiff = time.Duration(time.Now().UnixNano() - t.Sec*1000*1000*1000 - t.Nsec)
+}
+
+// WallTimeFromBootTime converts a time from bpf_ktime_get_boot_ns() to the
+// wall time with nano precision.
+//
+// Example:
+//
+//	fmt.Printf("Time: %s\n", WallTimeFromBootTime(ts).String())
+//
+// would display:
+//
+//	Time: 2022-12-15T16:49:00.452371948+01:00
+//
+// Shell command to convert the number to a date:
+//
+//	$ date -d @$(echo 1671447636499110634/1000000000|bc -l) +"%d-%m-%Y %H:%M:%S:%N"
+//	19-12-2022 12:00:36:499110634
+//
+// bpf_ktime_get_boot_ns was added in Linux 5.7. If not available and the BPF
+// program returns 0, just get the timestamp in userspace.
+func WallTimeFromBootTime(ts uint64) types.Time {
+	if ts == 0 {
+		return types.Time(time.Now().UnixNano())
+	}
+	return types.Time(time.Unix(0, int64(ts)).Add(timeDiff).UnixNano())
+}
+
+// detectBpfKtimeGetBootNs returns true if bpf_ktime_get_boot_ns is available
+// in the current kernel. False negatives are possible if BTF is not available.
+func detectBpfKtimeGetBootNs() bool {
+	btfSpec, err := btf.LoadKernelSpec()
+	if err != nil {
+		return false
+	}
+
+	enum := btf.Enum{}
+
+	err = btfSpec.TypeByName("bpf_func_id", &enum)
+	if err != nil {
+		return false
```

Comment:
Here I get the error:
```
*btf.Enum does not satisfy Type interface
```

It works when `TypeByName()` has a double pointer (`**btf.Enum`). Fixing...